storage/innobase/unittest/innodb_sync-t.cc

```diff
@@ -209,4 +209,7 @@ int main(int argc __attribute__((unused)), char **argv)
 
   ok(true, "sux_lock");
   sux.free();
+
+  my_end(MY_CHECK_ERROR);
+  return exit_status();
```

Comment:
Should this also be applied to the other InnoDB unit tests?

What are the symptoms of this memory leak? Something reported by AddressSanitizer?

Reply:
safemalloc reports a leak after the test runs:
```
==============================================================================

TEST                                      RESULT   TIME (ms) or COMMENT
--------------------------------------------------------------------------

worker[01] Using MTR_BUILD_THREAD 300, with reserved ports 19000..19029
unit.innodb_sync                         [ fail ]
        Test ended at 2024-12-06 09:44:04

CURRENT_TEST: unit.innodb_sync
1..4
ok 1 - srw_mutex
ok 2 - srw_lock
ok 3 - ssux_lock
ok 4 - sux_lock
Memory report from safemalloc
Warning: 2072 bytes lost at 0x14e809270, allocated by T@0 at 0x1025e32a8,
```
I've fixed some other `unit` tests as well (other PRs are up for those because the earliest branch they appear in differs from one another).

Reply:
(I only applied it to tests that demonstrate a problem on mac)